### PR TITLE
feat: [CGC-34] Change to DHIS2 UI

### PR DESCRIPTION
**Pull request by henrikmv**
[CGC-34](https://devotta.atlassian.net/browse/CGC-34?atlOrigin=eyJpIjoiNDhhMDk3NzM3OGE4NGQyMGE2ODA2YmQ2NTI5OGRmYWIiLCJwIjoiaiJ9)

- Change to DHIS2 UI for **dropdowns** and **menu button**
- CSS clean up

**NOTE:** This issue introduces a bug where the dropdown list goes upwards for Cypress testing. See bug issue [here](https://devotta.atlassian.net/browse/CGC-68?atlOrigin=eyJpIjoiNzViOGIzMzU3Y2JiNDA5ZWIxODgzNzg3MWM5YTVkZWEiLCJwIjoiaiJ9).
The workaround for now, is to use **Force click**.

[CGC-34]: https://devotta.atlassian.net/browse/CGC-34?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by petterav — COMMENTED** [review]
Well done changing to dhis ui! Look at the comments and i will take another review!

File: src/components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx [hunk 1]
```diff
@@ -21,60 +19,24 @@ export const ChartSettingsButton = ({
...
-                    <PopoverListItem
-                        label={i18n.t('Print')}
-                        icon={<PrintIcon />}
-                        onClick={handlePrintDocument}
-                    />
-                    <PopoverListDivider />
-                </PopoverList>
-            )}
-        </div>
+        <EllipsisButton
```
[the remark this answers is not in the document]
Comment (henrikmv) [reply src/components/GrowthChart/ChartSettingsButton/ChartSettingsButton.tsx:33]: The reason for moving the props out of the `EllipsisButton` is to make the conponent reusable.

File: src/Plugin.tsx [hunk 2]
```diff
@@ -88,38 +88,24 @@ const PluginInner = (propsFromParent: EnrollmentOverviewProps) => {
...
-                    width: '100vw',
-                    display: 'flex',
-                    margin: 0,
-                    padding: 0,
-                }}
-            >
-                <div
-                    style={{ width: '100%' }}
+        <QueryClientProvider client={queryClient}>
+            <div className='bg-white w-screen flex m-0 p-0'>
```
[the remark this answers is not in the document]
Comment (henrikmv) [reply src/Plugin.tsx:92]: Works properly for me, make a restart and tell me if the error still occur

File: cypress/component/ChartSelector/ChartSelectorDropdown/ChartSelectorDropdown.cy.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ describe('ChartSelectorDropdown', () => {
     const TestComponent = () => {
         const [title, handleItemChange] = useState('0 to 13 weeks');
         return (
-        // eslint-disable-next-line react/jsx-filename-extension
+            // eslint-disable-next-line react/jsx-filename-extension
             <ChartSelectorDropdown
```
[the remark this answers is not in the document]
Comment (henrikmv) [reply cypress/component/ChartSelector/ChartSelectorDropdown/ChartSelectorDropdown.cy.js:9]: There is no force click in this file. However, I think we should add it. It is an known error with the SingleSelect component in DHIS2, that the dropdown is going upwards. For know, we just solve the problem by jusing force click, so I have added it to the file. Agree?

File: src/components/GrowthChart/ChartSettingsButton/EllipsisButton/EllipsisButton.tsx [hunk 4]
```diff
@@ -1,22 +1,61 @@
...
+
+    const toggle = () => {
+        setActionsIsOpen((prev) => !prev);
+    };
+
+    return (
+        <div ref={anchorRef}>
+            {/* @ts-ignore */}
+            <Button
+                primary={primary}
```
[the remark this answers is not in the document]
Comment (henrikmv) [reply src/components/GrowthChart/ChartSettingsButton/EllipsisButton/EllipsisButton.tsx:38]: The reason for doing this is to make the `EllipsisButton` reusable. The props is sent from outside

File: src/components/GrowthChart/GrowthChartSelector/ChartSelector.tsx [hunk 5]
```diff
@@ -34,25 +34,28 @@ export const ChartSelector = ({
     };
 
     return (
-        <div className='flex flex-wrap w-full gap-2 text-sm'>
+        <div className='flex flex-wrap gap-2 items-center'>
```
[the remark this answers is not in the document]
Comment (henrikmv) [reply src/components/GrowthChart/GrowthChartSelector/ChartSelector.tsx:37]: No, this is only the placement of the diffrent dropdowns

File: src/components/GrowthChart/GrowthChartSelector/ChartSelectorDropdown/ChartSelectorDropdown.tsx [hunk 6]
```diff
@@ -17,48 +17,36 @@ export const ChartSelectorDropdown = ({
...
-                disabled
-                data-test={`${dataTest}-disabled-button`}
-            >
-                {title}
-            </button>
+}: ChartSelectorDropdownProps) => {
+    const tooltipContent = i18n.t('Gender is pre-selected based on the profile');
+    return (
+        isDisabled ? (
+            <Tooltip openDelay={500} closeDelay={50} content={tooltipContent}>
```
Comment (edvinstava) [inline src/components/GrowthChart/GrowthChartSelector/ChartSelectorDropdown/ChartSelectorDropdown.tsx:24]: Consider removing the const, as its only used in one place you might as well set the text string there. This also allows you to remove the return().

```suggestion
     (
        isDisabled ? (
            <Tooltip 
                openDelay={500} 
                closeDelay={50} 
                content={i18n.t('Gender is pre-selected based on the profile')}>
```

File: src/components/GrowthChart/GrowthChart.tsx [hunk 7]
```diff
@@ -79,44 +79,41 @@ export const GrowthChart = ({
...
+                    category={category}
+                    dataset={dataset}
+                    setCategory={setCategory}
+                    setDataset={setDataset}
+                    chartData={chartDataForGender}
+                    isDisabled={trackedEntityGender !== undefined}
+                    gender={gender}
+                    setGender={setGender}
+                />
+                <div className='relativ'>
```
Comment (edvinstava) [inline src/components/GrowthChart/GrowthChart.tsx:94]: "relativ" isn't a option with tailwindcss, either remove it or use "relative" as in the suggestion.
```suggestion
                <div className='relative'>
```

**Review by edvinstava — APPROVED** [review]
Well done Henrik, I just have 3 minor comments that you can take a look at. The dropdowns look great and generic! I will approve now as the requested changes are minor, but please take a look at my comments. 

Additionally, I get some changes to the yarn.lock file for the branch when running "yarn". Make sure to run yarn and upload any possible changes to the yarn.lock file. Furthermore, if you can run "yarn start", it will update the en.pot file as well.

Well done!

**Pull request merged**